src/ContentDecorator.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Widgets;
+
+use Yiisoft\View\Exception\ViewNotFoundException;
+use Yiisoft\View\WebView;
+use Yiisoft\Widget\Widget;
+
+/**
+ * ContentDecorator records all output between {@see begin()} and {@see end()]} calls, passes it to the given view file
+ * as `$content` and then echoes rendering result.
+ *
+ * ```php
+ * <?php ContentDecorator::begin()
+ *     ->viewFile('@app/views/layouts/base.php'),
+ *     ->params([]),
+ *     ->view($this),
+ * ]) ?>
+ *
+ * some content here
+ *
+ * <?php ContentDecorator::end() ?>
+ * ```
+ */
+class ContentDecorator extends Widget
+{
+    /**
+     * @var Aliases
+     */
+    private Aliases $aliases;
+
+    /**
+     * @var array the parameters (name => value) to be extracted and made available in the decorative view.
+     */
+    private array $params = [];
+
+    /**
+     * @var string the view file that will be used to decorate the content enclosed by this widget. This can be
+     * specified as either the view file path or alias path.
+     */
+    private ?string $viewFile = null;
+
+    /**
+     * @var WebView
+     */
+    private WebView $webView;
+
+    public function __construct(WebView $webview)
+    {
+        $this->webView = $webview;
+    }
+
+
+    public function init(): void
```

Comment:
We'd better come up with a more descriptive name...

Reply:
done.